internal/pkg/cli/secret_init.go

```diff
@@ -117,9 +149,75 @@ func (o *secretInitOpts) Ask() error {
 
 // Execute creates the secrets.
 func (o *secretInitOpts) Execute() error {
+	secretName := o.name
+	secretValues := o.values
+
+	if o.inputFilePath != "" {
+		var err error
+		secretName, secretValues, err = o.parseFile()
+		if err != nil {
+			return err
+		}
+	}
+
+	return o.putSecrets(secretName, secretValues)
+}
+
+func (o *secretInitOpts) putSecrets(secretName string, values map[string]string) error {
+	for envName, value := range values {
+		err := o.putSecret(secretName, envName, value)
+		if err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+func (o *secretInitOpts) putSecret(secretName, envName, value string) error {
+	client, err := o.configureSecretPutter(envName)
+	if err != nil {
+		return err
+	}
+
+	tags := make(map[string]string)
+	for k, v := range o.resourceTags {
+		tags[k] = v
+	}
+	tags[deploy.AppTagKey] = o.appName
+	tags[deploy.EnvTagKey] = envName
+
+	name := fmt.Sprintf(fmtSecretParameterName, o.appName, envName, secretName)
+	in := ssm.PutSecretInput{
+		Name:      name,
+		Value:     value,
+		Overwrite: o.overwrite,
+		Tags:      tags,
+	}
+
+	out, err := client.PutSecret(in)
+	if err != nil {
+		var targetErr *ssm.ErrParameterAlreadyExists
+		if errors.As(err, &targetErr) {
+			log.Infoln(fmt.Sprintf("Secret %s already exists. Do not overwrite.\n", name))
+			return nil
+		}
+		return fmt.Errorf("put secret %s in environment %s: %w", secretName, envName, err)
```

Comment:
Since the error message is already descriptive enough, we probably don't need this log message unless we are prompting them how to solve this issue (like an `override` flag or so) I would recommend updating the log msg with the solution.

Reply:
Do you mean the log message at line 201?

The log message is there because we don't return an error if the secret already exists. We just want to inform the user that it's this case and we didn't overwrite it.